aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -2360,36 +2356,45 @@ Tensor linalg_eigvalsh(const Tensor& input, c10::string_view uplo) {
     return values;
   }
 
-  squareCheckInputs(input);
-  checkUplo(uplo);
   ScalarType real_dtype = toValueType(input.scalar_type());
   Tensor values = at::empty({0}, input.options().dtype(real_dtype));
+  values = at::linalg_eigvalsh_outf(input, uplo, values);
+  return values;
+}
+
+Tensor& linalg_eigvalsh_out(const Tensor& input, c10::string_view uplo, Tensor& result) {
```

Comment:
Rewrote this function and registered it with `CPU, CUDA` dispatch key. This function is called in `linalg_eigvalsh` now and it allows to use `CompositeImplicitAutograd` key correctly for `linalg_eigvalsh`.